profiler/src/ProfilerEngine/Datadog.Profiler.Native.Windows/ETW/EtwEventsHandler.cpp

```diff
@@ -31,6 +34,12 @@ EtwEventsHandler::~EtwEventsHandler()
 
 void EtwEventsHandler::Stop()
 {
+    if (_pEventsFile != nullptr)
```

Comment:
I feel like there issues:
- a thread writing to a file being closed
- writing/closing/setting pointer to nullptr races

The fix I see is to add a `promise` 
I see a two promises/futures approach to fix that:
- In `Stop`, set the `_stopRequested` atomic variable to true. Wait on the `future` (given by the promise).
- In `OnConnect`, when getting out of the loop or finishing writing the file, set the value on the promise. This will unblock the thread executing `Stop`

example:
```
void EtwEventsHandler::Stop()
{
   _stopRequested = true;
   auto completed = _completionPromise.get_future();
   completed.wait();
   auto pEventsFile = std::exchange(_pEventsFile, nullptr);
   if (pEventsFile != nullptr)
   {
       fclose(pEventsFile);
   }
}
 ```
 and
 ```
 void EtwEventsHandler::OnConnect(HANDLE hPipe)
 {
    ...
    while (!_stopRequested.load())
    {
       ...
       // somewhere relevant
       _completionPromise.set_value();
       ...
 } 
 ```

Reply:
✅ implementation changed